src/OpenTelemetry/Metrics/Tags.cs

```diff
@@ -73,7 +74,7 @@ public Tags(KeyValuePair<string, object?>[] keyValuePairs)
             }
         }
 #else
-        for (int i = 0; i < ourKvps.Length; i++)
+        for (int i = 0; i < length; i++)
```

Comment:
Don't we want this to use one of the array's length properties? Would it be skipping the bounds check if we use a regular `int`?

Reply:
It does seem to elide using the local ya. Here's a demo: [SharpLab](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8DEA7ArgG30WHwFMACU7YsgWACgGABABnKYEYoBuBh8gewDM7AEzkAwv0EBvaYMEAHAE4BLAG6I4FANKkAngDVE+XKQAKiVcoA8nFjHIB7YACtSAYzgB+AHwBtAF1yAH09IxMzS2tkcgBecmxSAHdycONTCytbe0cXdy8/fw4WQN56BQV5SqYRVWw4cgBRfFUAE1IAFVJkOA4ACgBKaoUZSvGR8fJNZXIPJ1wG+PIWcqnKgDMnWf76xtVl1fIDm1D0yKyYgDoAGSoAczgAC25jgGo34Yr12Umf8mUpA200QswWyliCUBwLCBgyUWyyH8qjKfx+aPW80WjTeCXByCu8IoAEIEnhCORvOQOOQQCs1v9yABfPjff5MADscwWDQZ4yZ6zRaNqxyWLXaXR6cFEQzRY3+GMEM3IAGs4RdohDlrCIplNcg+ei2T9lVilglVoqBFsdntjodXidVeq9Yjbg9no6Pl9GeQ5MbfdCQWDcFqoUDnbqETFkaiA4yrZUzTi8aGCUTyKTEgR8JTqbT6YmWYx4+NOdzsYaFMXhXUxa0Ot1ekJZaW/YnlWqo5cw2cXdGIVWph3QeQyNhHk9ll2ifr3RPPYmR7Nkw7EzbyLslgcLY7yKdx5OvZ9E/7fQog8r8csgzONYjY0OFW3MTyU8404SLpmyTm8zS6UtF9BBrYCBHLZMn2ZaomSAA===)

Notice in "ElideTest1" there is `call 0x7ff9435f8f40` that is the thing that throws the range check failure. The other two don't have that.